Client/EngagementNotificationHelper.swift

```diff
@@ -0,0 +1,82 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Shared
+
+class EngagementNotificationHelper: FeatureFlaggable {
+    struct Constant {
+        #if MOZ_CHANNEL_FENNEC
+        // shorter time interval for development
+        static let timeUntilNotification: UInt64 = UInt64(60 * 2 * 1000) // 2 minutes in milliseconds
+        static let twentyFourHours: UInt64 = timeUntilNotification / 2 // 1 minutes in milliseconds
+        #else
+        static let timeUntilNotification: UInt64 = UInt64(60 * 60 * 48 * 1000) // 48 hours in milliseconds
+        static let twentyFourHours: UInt64 = UInt64(60 * 60 * 24 * 1000) // 24 hours in milliseconds
+        #endif
+        static let notificationId: String = "org.mozilla.ios.engagementNotification"
+    }
+
+    private var notificationManager: NotificationManagerProtocol
+    private var prefs: Prefs?
+    private var firstAppUse: Timestamp? {
+        prefs?.timestampForKey(PrefsKeys.KeyFirstAppUse)
```

Comment:
While this is swifty, the team a while ago opted to stick to using `return` in these types of situations simply due to the quick clarity of it.